modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/transfer/ProcessExport.java

```diff
@@ -0,0 +1,324 @@
+import org.wso2.carbon.pc.core.FolderZiper;
+import org.wso2.carbon.pc.core.ProcessCenterConstants;
+import org.wso2.carbon.pc.core.ProcessCenterException;
+import org.wso2.carbon.pc.core.ProcessStore;
+import org.wso2.carbon.pc.core.internal.ProcessCenterServerHolder;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProcessExport {
+    private List<String> exportedProcessList;
+    private static final Log log = LogFactory.getLog(ProcessExport.class);
+
+    /**
+     * @param processName
+     * @param processVersion
+     * @param exportWithAssociations
+     * @param user
+     * @return
+     * @throws Exception
+     */
+    public String initiateExportProcess(String processName, String processVersion, String exportWithAssociations,
+            String user) throws Exception {
+        exportedProcessList = new ArrayList<String>();
+        File exportsDir = new File(ProcessCenterConstants.PROCESS_EXPORT_DIR);
+        if (exportsDir != null) {
+            exportsDir.mkdirs();
+            Boolean exportWithAssociationsBool = Boolean.valueOf(exportWithAssociations);
+            try {
+                // save details about the exported zip and the core process
+                String exportRootPath =
+                        ProcessCenterConstants.PROCESS_EXPORT_DIR + "/" + processName + "-" + processVersion +
+                                ProcessCenterConstants.EXPORTS_DIR_SUFFIX + "/";
+                new File(exportRootPath).mkdirs();
+                exportProcess(exportRootPath, processName, processVersion, exportWithAssociationsBool, user);
+
+                //zip the folder
+                FolderZiper folderZiper = new FolderZiper();
+                String zipFilePath = ProcessCenterConstants.PROCESS_EXPORT_DIR + "/" + processName + "-" +
+                        processVersion + ProcessCenterConstants.EXPORTS_ZIP_SUFFIX;
+                folderZiper.zipFolder(exportRootPath, zipFilePath);
+                //encode zip file
+                String encodedZip = encodeFileToBase64Binary(zipFilePath);
+                //Finally remove the Imports folder and the zip file
+                FileUtils.deleteDirectory(exportsDir);
+                FileUtils.deleteDirectory(new File(zipFilePath));
+                return encodedZip;
+
+            } catch (Exception e) {
+                String errMsg = "Failed to export process:" + processName + "-" + processVersion;
+                log.error(errMsg, e);
+                throw new ProcessCenterException(errMsg, e);
```

Comment:
If you are throwing the exception, should not log here. It should be logged at the last position.